src/core/geometry/qgsgeometry.cpp

```diff
@@ -260,6 +260,47 @@ QgsGeometry QgsGeometry::collectGeometry( const QVector< QgsGeometry > &geometri
 
 QgsGeometry QgsGeometry::createWedgeBuffer( const QgsPoint &center, const double azimuth, const double angularWidth, const double outerRadius, const double innerRadius )
 {
+  if ( angularWidth >= 360.0 )
+  {
+    std::unique_ptr< QgsCompoundCurve > outerCc = qgis::make_unique< QgsCompoundCurve >();
```

Comment:
Have you looked into the QgsCircle class? That might be a better way to construct these. There's a constructor which takes a center point and radius, and then you can call toCircularString() to create the curve for the exterior/interior rings.

Reply:
Done